src/iceberg/transform_function.cc

```diff
@@ -148,14 +253,51 @@ Result<std::unique_ptr<TransformFunction>> YearTransform::Make(
 MonthTransform::MonthTransform(std::shared_ptr<Type> const& source_type)
     : TransformFunction(TransformType::kMonth, source_type) {}
 
-Result<ArrowArray> MonthTransform::Transform(const ArrowArray& input) {
-  return NotImplemented("MonthTransform::Transform");
-}
+Result<Literal> MonthTransform::Transform(const Literal& literal) {
+  assert(literal.type() == source_type());
+  if (literal.IsBelowMin() || literal.IsAboveMax()) {
+    return InvalidArgument(
+        "Cannot apply month transform to literal with value {} of type {}",
+        literal.ToString(), source_type()->ToString());
+  }
+  if (literal.IsNull()) [[unlikely]] {
+    return Literal::Null(iceberg::int32());
+  }
 
-Result<std::shared_ptr<Type>> MonthTransform::ResultType() const {
-  return iceberg::int32();
+  using namespace std::chrono;  // NOLINT
+  switch (source_type()->type_id()) {
+    case TypeId::kDate: {
+      auto value = std::get<int32_t>(literal.value());
```

Comment:
nit: I think we need also move all of these transform functions to a dedicated utility file and add exhaustive test cases. Could you create an issue for this?